lib/game_teams_manager.rb

```diff
@@ -13,122 +13,106 @@ def initialize(game_teams_path)
   end
 
   def assign_goals_by_team_hash
-      team_goals = Hash.new { |hash, key| hash[key] = [] }
-      @game_teams_array.each do |gameteam|
-        team_goals[gameteam.team_id] = []
-      end
-      @game_teams_array.each do |gameteam|
-        team_goals[gameteam.team_id] << gameteam.goals.to_i
-      end
-      team_goals
+    team_goals = Hash.new { |hash, key| hash[key] = [] }
+    @game_teams_array.each do |gameteam|
+      team_goals[gameteam.team_id] = []
+    end
+    @game_teams_array.each do |gameteam|
+      team_goals[gameteam.team_id] << gameteam.goals.to_i
+    end
+    team_goals
   end
 
-    def average_goals_by_team(team_goals)
-      team_goals.keys.each{|team|
-        team_goals[team] = (team_goals[team].sum.to_f/(team_goals[team].size)).round(2)}
+  def average_goals_by_team(team_goals)
+    team_goals.keys.each{|team|
+      team_goals[team] = (team_goals[team].sum.to_f/(team_goals[team].size)).round(2)}
       team_goals
     end
 
     def teams_max_by_average_goal(team_goals)
-      team_goals.values.max_by {|team| team}
+      team_goals.max_by {|k,v| v}.first
     end
 
     def teams_min_by_average_goal(team_goals)
-      team_goals.values.min_by {|team| team}
+      team_goals.min_by{|k,v| v}.first
```

Comment:
do min_by/max_by not only return one number?

Reply:
@crc2231 can you clarify your question?